### PR TITLE
Completed port for all working parts 

### PR DESCRIPTION
**Pull request by mayank18049**
1) dnet.c and pyrex.diff removed as they are intermediate files generated from setup.py and Makefiles
2) Added cython to setup.py.in
3) all test cases pass except 4 (2 firewall ,1 route,1 IP)

### CODE REVIEW

File: python/test.py [hunk 1]
```diff
@@ -151,25 +154,25 @@ def test_fw_loop(self):
...
+        self.assertTrue(self.intf, "couldn't open interface handle")
     def tearDown(self):
         del self.intf
 
     def test_intf_get(self):
-        lo0 = self.intf.get('lo0')
-        self.failUnless(lo0['name'] == 'lo0', "couldn't get loopback config")
-        self.failUnless(self.intf.get_src(dnet.addr('127.0.0.1')) == lo0,
+        lo0 = self.intf.get(interface)
+        self.assertTrue(lo0['name'] == 'lo', "couldn't get loopback config")
```
Comment (lnslbrty) [inline python/test.py:163]: That test will fail without reason if I change interface to 'eth0'.

File: python/test.py [hunk 2]
```diff
@@ -151,25 +154,25 @@ def test_fw_loop(self):
...
+        self.assertTrue(self.intf.get_src(dnet.addr('127.0.0.1')) == lo0,
```
Comment (lnslbrty) [inline python/test.py:164]: same here

File: python/test.py [hunk 3]
```diff
@@ -5,6 +5,9 @@
 sys.path.insert(0, './build')
 import dnet
 
+ip_address = '192.168.177.140'   #change according to subnet
```
Comment (lnslbrty) [inline python/test.py:8]: All tests referencing that address will fail on most machines without reason.
That is not how unittests should work.

File: python/test.py (hunk 2 above)
Comment (mayank18049) [reply python/test.py:165]: Added  a script which gets the local ip mac address and loopback_interface on any system.

**Comment by lnslbrty** [comment]
will review ASAP

**Comment by lnslbrty** [comment]
IMHO it would be better to split this bigfat MR into smaller MR's and also squash some commits.

**Pull request closed without merging**